File: freepascal/linux/HelloWorld_main.pas
program helloworld_main;

{$I+}

uses
            SysUtils;

procedure SafeWriteHello;
begin
            try
                        WriteLn('Hello World!');
                        if IOResult <> 0 then
                                    raise Exception.Create('WriteLn failed');
            except
                        on E: Exception do
                        begin
                                    WriteLn('Output error: ', E.Message);
                                    Halt(1);
                        end;
            end;
end;

begin
            SafeWriteHello;
end.

File: c_sharp/HelloWorld_main.cs
using System;

class HelloWorld_Main {
            static bool SafePrintHello() {
                        try {
                                    Console.Out.Flush();
                                    Console.WriteLine("Hello World!");
                                    Console.Out.Flush();
                                    return true;
                        } catch (Exception ex) {
                                    Console.Error.WriteLine($"Output error: {ex.Message}");
                                    return false;
                        }
            }

            static int Main() {
                        return SafePrintHello() ? 0 : 1;
            }
}
